cmd/k8s-bigip-ctlr/main.go

```diff
@@ -374,34 +373,32 @@ func getCredentials() error {
 			bigipURL = *credsDir + "/url"
 		}
 
-		usrBytes, err = ioutil.ReadFile(usr)
-		if err != nil {
-			log.Debug("No username in credentials directory, falling back to CLI argument")
-			if len(*bigIPUsername) == 0 {
-				return fmt.Errorf("BIG-IP username not specified")
+		setField := func(field *string, filename, fieldType string) error {
+			var fileBytes []byte
+			fileBytes, err = ioutil.ReadFile(filename)
```

Comment:
Do you mean to shadow err here? Because of the closure scope err is going to be the memory from outside this function. Might be safer to use := here to make sure we're using an err scoped to this function.

Reply:
go vet won't let me